command/git.go

```diff
@@ -55,56 +55,94 @@ type cloneOpts struct {
+		catcher.New("missing required owner")
 	}
 	if opts.repo == "" {
-		catcher.Errorf("missing required repo")
+		catcher.New("missing required repo")
 	}
-	if opts.method != "" && distro.ValidateCloneMethod(opts.method) != nil {
-		catcher.Errorf("method of cloning '%s' is invalid - must use legacy SSH or OAuth", opts.method)
+	if opts.location == "" {
+		catcher.New("missing required location")
+	}
+	if opts.method != "" {
+		catcher.Wrap(distro.ValidateCloneMethod(opts.method), "invalid clone method")
 	}
 	if opts.method == distro.CloneMethodOAuth && opts.token == "" {
 		catcher.New("cannot clone using OAuth if token is not set")
 	}
 	return catcher.Resolve()
 }
 
-func (opts cloneOpts) sshLocation() string {
+func (opts cloneOpts) sshLocationGitHub() string {
 	return fmt.Sprintf("git@github.com:%s/%s.git", opts.owner, opts.repo)
 }
 
-func (opts cloneOpts) httpLocation() string {
+func (opts cloneOpts) httpLocationGitHub() string {
 	return fmt.Sprintf("https://github.com/%s/%s.git", opts.owner, opts.repo)
 }
 
-// setLocation sets the location to clone from based on the method of cloning.
-func (opts *cloneOpts) setLocation() error {
+// setLocationGitHub sets the GitHub location to clone from.
+func (opts *cloneOpts) setLocationGitHub() error {
 	switch opts.method {
-	// No clone method specified is equivalent to using legacy SSH.
 	case "", distro.CloneMethodLegacySSH:
-		opts.location = opts.sshLocation()
+		opts.location = opts.sshLocationGitHub()
 	case distro.CloneMethodOAuth:
-		opts.location = opts.httpLocation()
+		opts.location = opts.httpLocationGitHub()
 	default:
-		return errors.Errorf("unrecognized clone method '%s' for this distro", opts.method)
+		return errors.Errorf("unrecognized clone method '%s'", opts.method)
 	}
 	return nil
 }
 
-func getCloneCommand(opts cloneOpts) ([]string, error) {
+// getProjectMethodAndToken returns the project's clone method and token. If
+// set, the project token takes precedence over global settings.
+func getProjectMethodAndToken(projectToken, globalToken, globalCloneMethod string) (string, string, error) {
+	if projectToken != "" {
+		token, err := parseToken(projectToken)
+		return distro.CloneMethodOAuth, token, err
+	}
+
+	switch globalCloneMethod {
+	// No clone method specified is equivalent to using legacy SSH.
+	case "", distro.CloneMethodLegacySSH:
+		return distro.CloneMethodLegacySSH, "", nil
+	case distro.CloneMethodOAuth:
+		if globalToken == "" {
+			return "", "", errors.New("cannot clone using OAuth if global token is empty")
+		}
+		token, err := parseToken(globalToken)
+		return distro.CloneMethodOAuth, token, err
+	}
+
+	return "", "", errors.Errorf("unrecognized clone method '%s'", globalCloneMethod)
+}
+
+// parseToken parses the OAuth token, if it is in the format "token <token>";
+// otherwise, it returns the token unchanged.
+func parseToken(token string) (string, error) {
+	if !strings.HasPrefix(token, "token") {
+		return token, nil
```

Comment:
Do you expect users to provide a token in some other format? If not, I think this should error.

Reply:
Some projects have created their own GItHub tokens under project variables and have not followed the "token <token>" format. They depend on the unformatted token as the "token" to `git.get_project`.